satpy/tests/reader_tests/test_fci_l1c_nc.py

```diff
@@ -509,6 +569,30 @@ def test_load_counts(self, reader_configs, fh_param,
             else:
                 numpy.testing.assert_array_equal(res[ch], 1)
 
+    @pytest.mark.parametrize("channel",list_total_channel)
+    @pytest.mark.parametrize("resolution",list_resolution)
+    def test_load_counts_af(self,FakeFCIFileHandlerAF_fixture,reader_configs,channel):
+        """Test loading with counts for AF files."""
+        expected_res_n = 1
+        fh_param = FakeFCIFileHandlerAF_fixture
+        type_ter = self._get_type_ter_AF(channel)
+        calibration = "counts"
+        res = self._get_res_AF(channel,fh_param,calibration,reader_configs)
+        assert expected_res_n == len(res)
+        for ch, grid_type in zip(fh_param["channels"][type_ter],
+                                 fh_param["channels"][f"{type_ter}_grid_type"]):
+            assert res[ch].shape == (GRID_TYPE_INFO_FOR_TEST_CONTENT[grid_type]["nrows"],
+                                     GRID_TYPE_INFO_FOR_TEST_CONTENT[grid_type]["ncols"])
+            assert res[ch].dtype == np.uint16
+            assert res[ch].attrs["calibration"] == calibration
+            assert res[ch].attrs["units"] == "count"
+            if ch == "ir_38":
+                numpy.testing.assert_array_equal(res[ch][-1], 1)
+                numpy.testing.assert_array_equal(res[ch][0], 5000)
+            else:
+                numpy.testing.assert_array_equal(res[ch], 1)
```

Comment:
this part is repeated many times in the different tests.. maybe it could go in a separate function to reduce code repetition?